lib/puppet/provider/package/apt.rb

```diff
@@ -147,7 +147,13 @@ def install
     end
 
     cmd += install_options if @resource[:install_options]
-    cmd << :install << str
+    cmd << :install
+
+    if should_use_source?
+      cmd << @resource[:source]
+    else
+      cmd << str
+    end
```

Comment:
Might be better to provide some info if the manifest has `ensure => 1.2.3, source => /path/to/packge`. In this case i would assume the version is ignored.

Reply:
I added check to allow only `:present` and `:installed` for ensure when using `source` parameter

Reply:
This could not be good enough since we may need a way to handle package upgrades

Reply:
what if we pass through ensure and in case source file was specified, we make sure the expected version was installed (this is what yum provider does in all cases):
https://github.com/puppetlabs/puppet/blob/c0e007a7b02b61a7264efba1d93ee7d0737e38b5/lib/puppet/provider/package/apt.rb#L165-L174

Reply:
I think this could work and it would be consistent.